cli/azd/pkg/account/manager.go

```diff
@@ -232,12 +232,15 @@ func (m *Manager) getDefaultSubscription(ctx context.Context) (*Subscription, er
 	subscriptionId := fmt.Sprint(configSubscriptionId)
 	subscription, err := m.azCli.GetAccount(ctx, subscriptionId)
 	if err != nil {
-		return nil, fmt.Errorf(
-			`the subscription id '%s' is either invalid or you no longer have access. 
-			Check your configuration with 'azd config list'. %w`,
-			subscriptionId,
-			err,
-		)
+		// if azd can't get the stored default account, it could means:
+		// - customer access was removed for the subscription
+		// - customer switched to a different account w/o access to the account
+		// - subscription was deleted.
+		// At any case, we don't want to fail on any of these cases, we should only return
+		msg := "is either invalid or you no longer have access. Check your configuration with 'azd config list'."
+		log.Printf("the subscription id %s %s Error: %s. Default subscription will be ignored.",
+			subscriptionId, msg, err.Error())
+		return nil, nil
```

Comment:
I still feel like this method `getDefaultSusbscription` should still fail and return an error if the subscription isn't accessible.  
Instead, have you considered updating the callers of this method to react to the error appropriatly?

https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/account/manager.go#L64
For `GetAccountDefaults()` it could return nil for subscription at this point

https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/account/manager.go#L89
For `GetSubscriptions()` it could just not have any subs set with the `IsDefault` flag.

Reply:
we would need to introduce a `type of error` which the callers can check.  I don't think we want to consider any type of error coming from the `getDefaultSusbscription()` as the same

Reply:
I could see a custom error as a nice to have but not required. It feels like this gives a false sense of security if this func doesn't throw an error.